crates/ruma-common/src/events/ignored_user_list.rs

```diff
@@ -26,6 +26,60 @@ impl IgnoredUserListEventContent {
     }
 }
 
+pub mod vec_as_map_of_empty {
+    use std::{collections::BTreeMap, marker::PhantomData};
+
+    use serde::{
+        de::{self, Deserialize, Deserializer},
+        ser::{Serialize, SerializeMap, Serializer},
+    };
+
+    use super::Empty;
+
+    /// Serialize the given `Vec<T>` as a map of `T => Empty`.
+    #[allow(clippy::ptr_arg)]
+    pub fn serialize<S, T>(vec: &Vec<T>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+        T: Serialize + Eq + Ord,
+    {
+        let map = serializer.serialize_map(Some(vec.len()))?;
+        for item in vec {
+            map.serialize_entry(item, Empty {})?;
+        }
+        map.end()
+    }
+
+    /// Deserialize an object and return the keys as a `Vec<T>`.
+    pub fn deserialize<'de, D, T>(deserializer: D) -> Result<Vec<T>, D::Error>
+    where
+        D: Deserializer<'de>,
+        T: Deserialize<'de> + Eq + Ord,
+    {
+        struct MapOfEmptyVisitor<T>(PhantomData<T>);
+        impl<'de, T> de::Visitor<'de> for MapOfEmptyVisitor<T> {
+            type Value = Vec<T>;
+
+            fn expecting(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                write!(f, "an object/map")
```

Comment:
shouldn't this be "an empty object/map"?

Reply:
No, `Empty` deserializes just fine from a non-empty object, plus the thing we're deserializing here is a map of user IDs to (usually) empty objects.